client/allocrunnerv2/taskrunner/task_runner.go

```diff
@@ -182,14 +184,14 @@ func NewTaskRunner(config *Config) (*TaskRunner, error) {
 		consulClient: config.Consul,
 		vaultClient:  config.VaultClient,
 		//XXX Make a Copy to avoid races?
-		state:        config.Alloc.TaskStates[config.Task.Name],
-		localState:   config.LocalState,
-		stateDB:      config.StateDB,
-		stateUpdater: config.StateUpdater,
-		ctx:          trCtx,
-		ctxCancel:    trCancel,
-		updateCh:     make(chan *structs.Allocation),
-		waitCh:       make(chan struct{}),
+		state:           config.Alloc.TaskStates[config.Task.Name],
+		localState:      config.LocalState,
+		stateDB:         config.StateDB,
+		stateUpdater:    config.StateUpdater,
+		ctx:             trCtx,
+		ctxCancel:       trCancel,
+		triggerUpdateCh: make(chan struct{}, 1), // see definition for cap
```

Comment:
I think a named variable might make this more clear instead of referencing via a comment.